net/nimble/host/src/ble_hs.c

```diff
@@ -313,7 +313,7 @@ ble_hs_sync(void)
 
     if (rc == 0) {
         rc = ble_hs_misc_restore_irks();
-        assert(rc == 0);
```

Comment:
Maybe we could add some info log saying that storage is not present?

Reply:
Thanks @rymanluk.  I agree that we should log something on failure.  I also think the `assert()` is a bit extreme, so I changed the code to log on failure but never crash.